File: app/src/main/java/ru/samtakoy/core/api/AppPreferencesImpl.java
package ru.samtakoy.core.api;

import android.content.Context;
import android.preference.PreferenceManager;

import javax.inject.Inject;

import static ru.samtakoy.core.api.AppPreferencesKeysKt.NEW_LEARNS_NOTIFICATION_MIN_UTC;
import static ru.samtakoy.core.api.AppPreferencesKeysKt.NEW_REPEATS_NOTIFICATION_MIN_UTC;
import static ru.samtakoy.core.api.AppPreferencesKeysKt.UNCOMPLETED_NOTIFICATION_MIN_UTC;

public class AppPreferencesImpl implements AppPreferences {


    /*private static final String NEW_REPEATS_NOTIFICATION_MIN_UTC = "NEW_REPEATS_NOTIFICATION_MIN_UTC";
    private static final String NEW_LEARNS_NOTIFICATION_MIN_UTC = "NEW_LEARNS_NOTIFICATION_MIN_UTC";
    private static final String UNCOMPLETED_NOTIFICATION_MIN_UTC = "UNCOMPLETED_NOTIFICATION_MIN_UTC";*/


    @Inject
    Context mContext;

    @Inject
    public AppPreferencesImpl() {
        //mContext = ctx;
    }

    // --

    public long getNewRepeatsNotificationMinUtc() {
        return PreferenceManager.getDefaultSharedPreferences(mContext)
                .getLong(NEW_REPEATS_NOTIFICATION_MIN_UTC, 0);
    }

    public void setNewRepeatsNotificationMinUtc(long value) {
        PreferenceManager.getDefaultSharedPreferences(mContext)
                .edit()
                .putLong(NEW_REPEATS_NOTIFICATION_MIN_UTC, value)
                .apply();
    }

    // --

    public String getString(String key, String defValue) {
        return PreferenceManager.getDefaultSharedPreferences(mContext)
                .getString(key, defValue);
    }

    public void setString(String key, String value) {
        PreferenceManager.getDefaultSharedPreferences(mContext)
                .edit()
                .putString(key, value)
                .apply();
    }

    // --

    public long getNewLearnsNotificationMinUtc() {
        return PreferenceManager.getDefaultSharedPreferences(mContext)
                .getLong(NEW_LEARNS_NOTIFICATION_MIN_UTC, 0);
    }

    public void setNewLearnsNotificationMinUtc(long value) {
        PreferenceManager.getDefaultSharedPreferences(mContext)
                .edit()
                .putLong(NEW_LEARNS_NOTIFICATION_MIN_UTC, value)
                .apply();
    }

    // --


    // --

    public long getUncompletedNotificationMinUtc() {
        return PreferenceManager.getDefaultSharedPreferences(mContext)
                .getLong(UNCOMPLETED_NOTIFICATION_MIN_UTC, 0);
    }

    public void setUncompletedNotificationMinUtc(long value) {
        PreferenceManager.getDefaultSharedPreferences(mContext)
                .edit()
                .putLong(UNCOMPLETED_NOTIFICATION_MIN_UTC, value)
                .apply();
    }

    // --

}

File: app/src/main/kotlin/ru/samtakoy/features/preferences/data/AppPreferencesImpl.kt
package ru.samtakoy.features.preferences.data

import android.content.Context
import android.preference.PreferenceManager

/** TODO переделать (префы с привязкой к имени файла всесто app) */
class AppPreferencesImpl(
    private val context: Context
) : AppPreferences {

    var newRepeatsNotificationMinUtc: Long
        // --
        get() = PreferenceManager.getDefaultSharedPreferences(context)
            .getLong(NEW_REPEATS_NOTIFICATION_MIN_UTC, 0)
        set(value) {
            PreferenceManager.getDefaultSharedPreferences(context)
                .edit()
                .putLong(NEW_REPEATS_NOTIFICATION_MIN_UTC, value)
                .apply()
        }

    // --
    override fun getString(key: String, defValue: String?): String? {
        return PreferenceManager.getDefaultSharedPreferences(context)
            .getString(key, defValue)
    }

    override fun setString(key: String, value: String?) {
        PreferenceManager.getDefaultSharedPreferences(context)
            .edit()
            .putString(key, value)
            .apply()
    }

    var newLearnsNotificationMinUtc: Long
        // --
        get() = PreferenceManager.getDefaultSharedPreferences(context)
            .getLong(NEW_LEARNS_NOTIFICATION_MIN_UTC, 0)
        set(value) {
            PreferenceManager.getDefaultSharedPreferences(context)
                .edit()
                .putLong(NEW_LEARNS_NOTIFICATION_MIN_UTC, value)
                .apply()
        }

    // --
    // --
    fun getUncompletedNotificationMinUtc(): Long {
        return PreferenceManager.getDefaultSharedPreferences(context)
            .getLong(UNCOMPLETED_NOTIFICATION_MIN_UTC, 0)
    }

    override fun setUncompletedNotificationMinUtc(value: Long) {
        PreferenceManager.getDefaultSharedPreferences(context)
            .edit()
            .putLong(UNCOMPLETED_NOTIFICATION_MIN_UTC, value)
            .apply()
    }
}
